storey/sql_driver.py

```diff
@@ -108,23 +114,12 @@ async def _get_specific_fields(self, key: str, table, attributes: List[str]):
     def supports_aggregations(self):
         return False
 
-    def _get_where_clause(self, key, sql_table):
-        where_clause = ""
-        key = self._extract_list_of_keys(key)
-        for i in range(len(self._primary_key)):
-            if i != 0:
-                where_clause += " and "
-            if sql_table.columns[self._primary_key[i]].type.python_type == str:
-                where_clause += f'{self._primary_key[i]}="{key[i]}"'
-            else:
-                where_clause += f"{self._primary_key[i]}={key[i]}"
-        return where_clause
-
-    def _update_by_key(self, key, data, table):
-        where_clause = self._get_where_clause(key, table)
-        update_clause = " ,".join([f'{key}="{value}"' for key, value in data.items() if key not in self._primary_key])
-        sql_statement = f"UPDATE {table} SET {update_clause} where {where_clause}"
-        self._sql_connection.execute(sql_statement)
+    def _update_by_key(self, key, data, sql_table):
+        self._sql_connection.execute(
+            db.update(sql_table)
+            .values({getattr(sql_table.c, k): v for k, v in data.items() if k not in self._primary_key})
+            .where(db.and_(getattr(sql_table.c, self._primary_key[i]) == key[i] for i in range(len(self._primary_key))))
```

Comment:
I see a warning for `self._primary_key[i]` in PyCharm:
```
Expected type 'str', got 'int' instead 
```
Seems like there is a case where that can happen and that it's probably not correct.